### DISCUSSION TITLE
Proposal: NoSQL Database Support in Eloquent ORM

### DISCUSSION THREAD

**Discussion by Raza9798** [laravel/framework#50681] [Ideas]
# Proposal: NoSQL Database Support in Eloquent ORM

## Overview

While Laravel's Eloquent ORM provides a robust solution for working with relational databases, there's a noticeable gap when it comes to NoSQL database support. This proposal suggests integrating NoSQL database support into Laravel's Eloquent ORM to provide developers with a seamless experience when working with databases like MongoDB or Cassandra.

## Features to Consider

1. **Dynamic Schema Support:**
   - Allow Eloquent ORM to handle the dynamic schemas of NoSQL databases, providing more flexibility in data modeling.

2. **Scalability Enhancements:**
   - Leverage the scalability features of NoSQL databases to handle large volumes of data and high-speed transactions efficiently.

3. **Hybrid Database Models:**
   - Support hybrid models capable of interacting with both SQL and NoSQL databases within the same application, offering versatility in data storage and retrieval.

## Comparison with Other Frameworks

Frameworks like Django have packages extending their ORM capabilities to work seamlessly with NoSQL databases, positioning Laravel to catch up with this trend.

## Potential Benefits

- Improved performance and scalability for Laravel applications, especially those dealing with large datasets or high transaction rates.
- Greater flexibility in data modeling and storage options, accommodating diverse application requirements.
- Consistency in development experience for Laravel developers accustomed to working with Eloquent ORM.

## Discussion Points

- Feasibility: Assess the technical challenges and feasibility of integrating NoSQL support into Eloquent ORM.
- Compatibility: Ensure compatibility with existing Laravel functionalities and packages.
- Community Interest: Gauge the interest and demand within the Laravel community for such a feature.
- Implementation Details: Discuss the potential implementation strategies and any potential trade-offs.
- Documentation and Support: Consider the need for comprehensive documentation and support resources for developers adopting this feature.

## Next Steps

- Collaborate on the design and implementation of this feature.
- Gather feedback from the Laravel community to refine the proposal.
- Develop a roadmap for integrating NoSQL support into Eloquent ORM.

Let's discuss the potential of bringing NoSQL database support to Laravel's Eloquent ORM and empower developers with more choices in database technologies. Your insights and ideas are more welcome

**Comment by ghost** [comment]
Hi. mongoDb is supported via jenssegers/mongodb. Maybe that will help.

**Reply by bridgeyuwa** [reply]
Stumbled across **spatie/laravel-schemaless-attributes**

Solved my laravel EAV issues.
